### PR TITLE
case-insenstive matching (for newer haproxy version) and proper collectd types

### PR DESCRIPTION
**Pull request by tux21b**
We had to make some trivial fixes for our setup. "MaxConn", "MaxPipes", etc. is now called "Maxconn", "Maxpipes", ... in haproxy and therefore the script wasn't able to find those values anymore. This patch converts all keys to lowercase internally for matching purposes.

Also, we had to change the collectd types from counter to derive (gouge:0:u) to handle haproxy restarts correctly.

Feel free to contact us if you have any questions.

### CODE REVIEW

File: haproxy.py [hunk 1]
```diff
@@ -18,44 +18,44 @@
 RECV_SIZE = 1024
 METRIC_TYPES = {
     'MaxConn': ('max_connections', 'gauge'),
-    'CumConns': ('connections', 'counter'),
-    'CumReq': ('requests', 'counter'),
+    'CumConns': ('connections', 'derive'),
```
Comment (vzhabiuks) [inline haproxy.py:21]: Hi Christoph
Could you explain why did you use derive types instead of counter?

**Comment by mdubbyap** [comment]
Hi Christophe, can you take a look at https://github.com/signalfx/collectd-haproxy/pull/4 and see if it satisfies all your needs?  We took a slightly different tack. If you're satisfied, please close this PR.  Thanks!

**Comment by tux21b** [comment]
@mdubbyap i just took a quick look at the other commit. The relevant parts (i.e. using proper collectd types and using lower-case keys for the comparison) seem to have been copied from this PR, so I am happy with that. Interestingly, the same commit (?) also seems to contain some refactoring. That's not a bad thing but I would have liked a separate commit for that.

The thing that worries me is that the refactoring isn't compatible with the old version. I don't mind how the values are called, as long as they keep their name. Updating to the new version would require renaming all values in the front-ends and monitoring checks, and even worse, would require TWICE the storage in the back-end. Therefore an update is not possible. Such changes should be avoided at all costs unless they are really necessary, and then they should definitely be done in a separate commit including an announcement and a new major release...

**Comment by vzhabiuks** [comment]
Hi Christoph

I apologize for backward incompatible change. Let me post another PR to
revert metric name changes

With many thanks,
Volodymyr

**Comment by vzhabiuks** [comment]
https://github.com/signalfx/collectd-haproxy/pull/5/files

Testing this right now

**Comment by mdubbyap** [comment]
I've merged Vlod's reversions. Sorry about that Christophe.   Thanks again for submitting your PR.

matthew.

**Pull request closed without merging**